enhancements/multi-arch/dynamic-imagestream-importmode-setting.md

```diff
@@ -0,0 +1,469 @@
+
+``` 
+type ImageSpec struct {
+...
+...
+// imagestreamImportMode controls the import
+// mode behaviour of imagestreams. It can be set
+// to `Legacy` or `PreserveOriginal`. The
+// default behaviour is to default to Legacy.
+// If this value is specified, this setting is
+// applied to all imagestreams which do not
+// have the value set.
+// +optional
+ImageStreamImportMode imagev1.ImportModeType
+`json:"imagestreamImportMode,omitempty" }`
+```
+```
+type ImageStatus struct {
+...
+...
+// imagestreamImportMode controls the import
+// mode behaviour of imagestreams. It can be set
+// to `Legacy` or `PreserveOriginal`. The
+// default behaviour is to default to Legacy.
+// If this value is specified, this setting is
+// applied to all new imagestreams which do not
+// have the value set.
+// +optional
+ImagestreamImportMode imagev1.ImportModeType
+`json:"imagestreamImportMode,omitempty"
+}
+```
+- openshiftcontrolplane/v1/: Add
+  `ImagestreamImportMode` config flag to
+  ImagePolicyConfig struct.
+
+```
+type ImagePolicyConfig {
+...
+...
+// imagestreamImportMode controls the import
+mode behaviour of imagestreams. It can be set
+to `Legacy` or `PreserveOriginal`. The
+// default behaviour is to default to Legacy.
+// If this value is specified, this setting is
+// applied to all imagestreams which do not
+// have the value set.
+// +optional
+ImagestreamImportMode imagev1.ImportModeType
+`json:"imagestreamImportMode,omitempty"`
+
+}
+```
+- An example CR for `image.config.openshift.io`
+  would look like below if the import mode is set:
+```
+apiVersion: config.openshift.io/v1
+kind: Image
+metadata:
+  name: cluster
+  ...
+  ...
+spec:
+  imagestreamImportMode: PreserveOriginal
+status:
+   internalRegistryHostname: image-registry.openshift-image-registry.svc:5000
+   imagestreamImportMode: PreserveOriginal
+  ...
+  ...
+
+```
+- The ClusterVersion status field CR would look
+  like below after CVO has inferred the payload
+  type:
+```
+apiVersion: v1
+items:
+- apiVersion: config.openshift.io/v1
+  kind: ClusterVersion
+  spec:
+    channel: candidate-4.15
+    desiredUpdate:
+      force: false
+      version: 4.15.0
+  status:
+    availableUpdates:
+    ...
+    capabilities:
+    ...
+    conditions:
+    ...
+    desired:
+    ...
+    history:
+    ...
+    payloadArchitecture: amd64  # could be arm64/ppc64le/s390x/multi
+    ...
+``` 
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+N/A
+
+#### Standalone Clusters
+
+N/A
+
+#### Single-node Deployments or MicroShift
+
+N/A
+
+### Implementation Details/Notes/Constraints
+
+- CVO: The cluster version operator would expose a
+  field in its status indicating the
+  architecture/type of the payload.
+    - If architecture, then it could be `amd64`,
+      `arm64`, `ppc64le`, `s390x` or `multi`.
+    - If type, then it could be `single` or
+      `multi`.
+- Cluster-image-registry-operator: the operator
+  sets the `imagestreamImportMode` field in the
```

Comment:
nit: `imagestreamImportMode` -> `imageStreamImportMode`?  Obviously folks will understand either way, I'm just surprised to not see the `Stream` portion capitalized.

Reply:
yup..i'll change those...because "imagestreams" is generally written as a single word, but for API purposes, it needs to be capitalized.